src/hotspot/share/opto/memnode.cpp

```diff
@@ -1581,23 +1650,52 @@ bool LoadNode::can_split_through_phi_base(PhaseGVN* phase) {
     base = base->in(1);
   }
 
-  if (req() > 3 || base == nullptr || !base->is_Phi()) {
+  if (req() > 3 || !can_split_through_phi_helper(base, mem)) {
     return false;
   }
 
-  if (!mem->is_Phi()) {
-    if (!MemNode::all_controls_dominate(mem, base->in(0))) {
-      return false;
-    }
-  } else if (base->in(0) != mem->in(0)) {
-    if (!MemNode::all_controls_dominate(mem, base->in(0))) {
-      return false;
+  if (nested) {
+    for (uint i = 1; i < base->req(); i++) {
+      if (base->in(i)->is_Phi()) {
+        // base->in(i) is the parent phi node for base node.
+        Node *mem_node_for_load_after_opt = get_memory_node_for_nestedphi_after_split(phase, base, i);
```

Comment:
```suggestion
        Node* mem_node_for_load_after_opt = get_memory_node_for_nestedphi_after_split(phase, base, i);
```
Drive by style comment